docs/source/developer/ui_components.rst

```diff
@@ -0,0 +1,337 @@
+.. _ui_components:
+
+Reusing JupyterLab UI
+=====================
+
+The `@jupyterlab/ui-components <http://jupyterlab.github.io/jupyterlab/ui-components/index.html>`__
+package provides UI elements that are widely used in JupyterLab core,
+and that can be reused in your own extensions.
+
+For example, all of the icons in JupyterLab core can be reused via
+``LabIcon``. You can also use ``LabIcon`` to create your own custom
+icons that will be able to automatically change color to match the
+current JupyterLab theme.
+
+.. contents:: Explainer docs
+   :local:
+   :depth: 1
+
+``LabIcon`` - set up and render icons
+-------------------------------------
+
+``LabIcon`` is the icon class used by JupyterLab, and is part of the new
+icon system introduced in JupyterLab v2.0.
+
+Background
+~~~~~~~~~~
```

Comment:
Could we move the background to the end? I really appreciate it, but think it's probably best to just show ppl how to use it first and foremost, and then they can read about the history after.

Reply:
sure